pyvista/core/filters/poly_data.py

```diff
@@ -2755,6 +2755,113 @@ def extrude_rotate(
             return self
         return output
 
+    def extrude_trim(
+        self,
+        direction,
+        trim_surface,
+        extrusion="boundary_edges",
+        capping="intersection",
+        inplace=False,
+        progress_bar=False,
+    ):
+        """Extrude polygonal data trimmed by a surface.
+
+        This generates polygonal data on output.
+
+        The input dataset is swept along a specified direction forming a
+        "skirt" from the boundary edges 2D primitives (i.e., edges used
+        by only one polygon); and/or from vertices and lines. The extent
+        of the sweeping is defined where the sweep intersects a
+        user-specified surface.
+
+        Parameters
+        ----------
+        direction : numpy.ndarray or sequence
+            Direction vector to extrude.
+
+        trim_surface: pyvista.PolyData
+            Surface which trims the surface.
+
+        extrusion : str or int, optional
+            Control the strategy of extrusion. One of the following:
+
+            * ``"boundary_edges"`` or ``0``
+            * ``"all_edges"`` or ``1``
+            The default is "boundary_edges".
+
+        capping : str or int, optional
+            Control the strategy of capping. One of the following:
+
+            * ``"intersection"`` or ``0``
+            * ``"minimum_distance"`` or ``1``
+            * ``"maximum_distance"`` or ``2``
+            * ``"average_distance"`` or ``3``
+            The default is "intersection".
+
+        inplace : bool, optional
```

Comment:
If this can return non PolyData, we would need to handle the case where the user also selects `inplace=True` to give a sensible error message.

Reply:
Return type is always a `PolyData`.